scripts/simplify_network.py

```diff
@@ -522,6 +523,64 @@ def consense(x):
     return clustering.network, clustering.busmap
 
 
+def merge_isolated_buses(n, drop_p_threshold=0):
+    generators_sum_origin = n.generators.p_nom.sum()
+    load_sum_origin = n.loads_t.p_set.sum().sum()
+
+    # duplicated sub-networks mean that there is at least one interconnection between buses
+    i_islands = n.buses[
+        (~n.buses.duplicated(subset=["sub_network"], keep=False))
+        & (n.buses.carrier == "AC")
+    ].index
+
+    load_sum_isol_origin = n.loads_t.p_set[i_islands].sum().sum()
+
+    # disregard buses having very small attached load when rearranging
+    loads_t_avr = n.loads_t.p_set.mean(axis=0)
+    i_islands_large = i_islands[loads_t_avr[i_islands] >= drop_p_threshold]
```

Comment:
does this work? the two elements may have different length

Reply:
It works as `i_islands_large` is an index created right here. So, we can subset `i_islands ` with a boolean series and write into `i_islands_large` despite both `i_*` usually have different lengths. Although, agree that in case of dataframes such assignment could lead to troubles

Reply:
with the i_islands check it should work, the [i_islands] does the trick :)